File: 2022/01/06/main.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        l = {}
        for t in trips:
            num, start, end = t[0], t[1], t[2]
            if start in l:
                l[start] += num
            else:
                l[start] = num
            if end in l:
                l[end] -= num
            else:
                l[end] = num

        p = 0
        for c in l:
            p += c
            if p > capacity:
                return False
        return True

File: 2022/01/06/main.go
package main

func carPooling(trips [][]int, capacity int) bool {
	l := [1001]int{}
	for _, t := range trips {
		num, start, end := t[0], t[1], t[2]
		l[start] += num
		l[end] -= num
	}

	p := 0
	for _, c := range l {
		p += c
		if p > capacity {
			return false
		}
	}
	return true
}
